src/ORM/Table.php

```diff
@@ -1011,6 +1011,14 @@ public function findAll(Query $query, array $options)
      */
     public function findList(Query $query, array $options)
     {
+        if (!isset($options['keyField']) && !isset($options['valueField'])) {
+            $select = $query->clause('select');
+            if ($select && count($select) <= 2) {
+                $options['keyField'] = array_shift($select);
+                $options['valueField'] = array_shift($select) ?: $options['keyField'];
```

Comment:
In any of those cases the select has been issued, so if it fails, then because he did not also join (contain) that association.
For all other cases one **must** manually set keyField and valueField of course.
This here only covers the trivial case of having two simple selects but no keyField/valueField selected AND otherwise the *wrong* fields selected silently.